File: LENGUAJES/PHP/Ejercicio2.php
<?php
// BUCLES WHILE
function obtenerNumero() {
    while (true) {
        echo "Ingrese un número para conocer la tabla de multiplicar: ";
        $input = trim(fgets(STDIN));
        
        // Validar que la entrada sea un número entero
        if (is_numeric($input) && intval($input) == $input) {
            return intval($input);
        } else {
            echo "Error, por favor ingrese un número entero.\n";
        }
    }
}

function mostrarTabla($num) {
    echo "\nTabla de multiplicar del $num:\n";
    for ($i = 1; $i <= 10; $i++) {
        echo "$num x $i = " . ($num * $i) . "\n";
    }
}

function main() {
    while (true) {
        $numero = obtenerNumero();
        mostrarTabla($numero);
        
        echo "\n¿Desea calcular otra tabla? (s/n): ";
        $continuar = trim(fgets(STDIN));
        
        if (strtolower($continuar) !== 's') {
            echo "Gracias por usar el programa. ¡Hasta luego!\n";
            break;
        }
    }
}
main();
?>

File: LENGUAJES/PYTHON/Ejercicio2.py
#Bucles WHILE

def obtener_numero():
    while True:
        try:
            num = int(input("Ingrese un número para conocer la tabla de multiplicar: "))
            return num
        except ValueError:
            print("Error, por favor ingrese un número entero.")

def mostrar_tabla(num):
    print(f"\nTabla de multiplicar del {num}:")
    for i in range(1, 11):
        print(f"{num} x {i} = {num * i}")

def main():
    while True:
        numero = obtener_numero()
        mostrar_tabla(numero)
        
        continuar = input("\n¿Desea calcular otra tabla? (s/n): ").strip().lower()
        if continuar != 's':
            print("Gracias por usar el programa. ¡Hasta luego!")
            break
main()